### DISCUSSION TITLE
Why I will stay with paperless-ng

### DISCUSSION THREAD

**Discussion by mtettke123** [jonaswinkler/paperless-ng#1526] [Feature Requests]
Hi out there,

I've tested quite a couple of DMS for my personal use and decided to stay with paperless-ng for the future (don't say how long, though, to keep options open :-) ). Here is why:

- very easy to install and use (docker-compose), quite good "handable" and easy to understand (both from administration and user view - I don't like overly complex java monsters)
- very easy to use, not overcomplex and responsible (in the meaning of time) web interface (ever tried ecoDMS? cruel!)
- files are stored "plain" in the filesystem (main reason to stay with paperless-ng as I sync them periodically to my owncloud and have a clear exit strategy if anything goes wrong!)
- nice and friendly community and creator :-)
- edit: The startpage showing searches is also great!

Having said this and after having just tested docspell, here are things I really liked about docspell and really wish paperless-ng would adopt:
- custom fields with the ability to sum up (I can just search last quotes - a.e. for service of my car - and get the sum of costs - great), see [here](https://gitanswer.com/paperless-ng-custom-metadata-python-778180255) also)
- "concerning", "direction" classifiers (altough could be done with Tags, not thaaaat important)
- "due date" - it makes sense to have a possibility to schedule a notification when adding documents
- folders with permissions - and in general, a permission and multi user management - although I completely understand that's a big task (see [here](https://github.com/jonaswinkler/paperless-ng/discussions/52) also - maybe files could be stored just with another classifier?)
- 2FA methods for login: I would love to have my DMS reachable on my public server - but would never expose a service that does not offer 2FA - there is possible very sensitive information in a DMS (see [here](https://gitanswer.com/feature-request-2-factor-authentication-python-paperless-ng-768581103) also); [container security is another point](https://github.com/jonaswinkler/paperless-ng/issues/1448).
- colors of the web interface (I don't like that paperless-ng green, but hej, that's just my personal view and not crucial) and look and feel
- docspell seems to be very well customizable by the user
- the way docspell extracts date from a document and shows all possibilities seems to be more comfortable than paperless-ng inserting just one date and not showing other possibilities
- notifcation options (new document added, due date reached, ....)

As there are issues for many of my points, I will not add this text as another one ([here](https://github.com/jonaswinkler/paperless-ng/issues/53) is already a good overview). I really hope paperless-ng can include some of them, as the basics of paperless-ng (mainly storing files in a customable structure in the filesystems, easyness of use and maintanance) are soooo great!

So, keep up the good work!

edit 2022-06-09: so happy that paperless is developing great with the [ngx Team](https://github.com/paperless-ngx/paperless-ngx). So enthusiastic people, so many good ideas. Great to use a well maintained piece of software. Although many of my wishes still say ... wishes, there are some very interesting features coming or in discussion, a.e.:

- [Feature ability to consume mails and eml files](https://github.com/paperless-ngx/paperless-ngx/pull/848)
- [Add MariaDB support](https://github.com/paperless-ngx/paperless-ngx/pull/543)
- [Expiration Date](https://github.com/paperless-ngx/paperless-ngx/pull/194)
- [Add a searchable comment field](https://github.com/paperless-ngx/paperless-ngx/discussions/687)
- [Custom metadata for document types](https://github.com/paperless-ngx/paperless-ngx/discussions/141)

So, still happy with my decision!

**Comment by lippoliv** [comment]
> "due date" - it makes sense to have a possibility to schedule a notification when adding documents

I like that field and to have a notification at a given date. This should also have a way to be defined as a view "overdue documents" and the user should be able to mark the document as "Done" (maybe by clear the due date field).

> the way docspell extracts date from a document and shows all possibilities seems to be more comfortable than paperless-ng inserting just one date and not showing other possibilities

Showing multiple found dates (by order) would be very helpfull indeed. I see it depends on the correspondent, but I have some, where paperless does use the "first" found date, but for that correspondant, it would be the second one. So I have to type.
If it would show up, I could just click it, nice :)

> notifcation options (new document added, due date reached, ....)

Also I like that Idea. I am really bad on document management, if there is no trigger. Currently I have set up a todo in my tasks management app for each sunday morning to check paperless and probably import documents from some external sources.
To keep the effort low, I use mail configuration and some rules, so at least some regular documents I do receive by e-mail for the paperless import.

For those, having some notification (an email would already be enough), would be interesting. 
What channels would users expect here to be supported?

**Reply by mtettke123** [reply]
Since I've written the text above, I've spent some hours with paperless-ng and docspell. While I really like the look and feel of the docspell interface and the advanced features, the above is still true. If you want your family to use a software, it should be as easy as possible, and that is absolutely true for paperless-ng. 

paperless-ng is so easy to use, there is nearly no need to learn something. doscpell on the other hand has some so-great-features, but I still struggle to find the correct button sometimes. That would probably not work for my wife (lower [WAF](https://en.wikipedia.org/wiki/Wife_acceptance_factor)).

And, as I sync the paperless-ng files directly to my owncloud, I've them always available anywhere. Thats cool. And the android apps really work good, although they are still work in progress.

So, I'm still hoping some more features come to paperless-ng, without making it much more cmplex or lowering the WAF.

**Comment by Cantello** [comment]
I was in the same boat and took a different direction than you. Have been using paperless-ng for nearly a year now, indexed ~1000 documents and I was sorely lacking one feature: invoice numbers (or any custom metadata for documents). That was the sole reason I switched and I still like most things better in paperless (ease of use, web UI, files stored directly on disk) but this one feature makes it up for me.

**Reply by Cantello** [reply]
Aaand I came back. Clunky web-UI and lack of ease of use weighed heavier than the custom metadata. I simply use the title field to enter invoice numbers now when I really need them (vendors with multiple invoices in a shorter time-frame) and it works now. Happy with paperless (ng or ngx in the future).

**Comment by ybizeul** [comment]
I too, wanted 2FA for that kind of things before I could expose to internet, and I finally chose to proxy paperless-ng via [Authelia](https://github.com/authelia/authelia), and very happy with it. Works perfectly fine with Apple Keychain 2FA and integrates with Active Directory. You can set Paperless to auto-login to the user that has just been authenticated by Authelia.

**Comment by simonschllng** [comment]
Thanks for testing other options for us. I did not do a research, paperless-ng felt just right.
I would extend your wish for 
> custom fields with the ability to sum up

by a way to handle contracts. There are a lot of documents related to a contract. The contract itself, payment notes, changes, invoices, warnings etc. And there is meta data like the contract number or e.g. payment terms. While I failed in the past with a full bookkeeping solution for home banking, having an overview over my contracts would be very beneficial for me.
